ev3/legoev3_motor.c

```diff
@@ -1403,31 +1050,71 @@ static unsigned legoev3_motor_get_commands (void *context)
 }
 
 static int legoev3_motor_send_command(void *context,
-				      struct tacho_motor_params *params,
-				      enum tacho_motor_command command)
+					struct tacho_motor_params *params,
+					enum tacho_motor_command command)
 {
 	struct legoev3_motor_data *ev3_tm = context;
 
-	ev3_tm->active_params = *params;
+	switch (command) {
+	case TM_COMMAND_RUN_TO_ABS_POS:
+			ev3_tm->position_sp = params->position_sp;
+			break;
+	case TM_COMMAND_RUN_TO_REL_POS:
+			ev3_tm->position_sp += params->position_sp;
+			break;
+	case TM_COMMAND_RESET:
+			legoev3_motor_reset(ev3_tm);
+			break;
+	case TM_COMMAND_RUN_FOREVER:
+	case TM_COMMAND_RUN_TIMED:
+	case TM_COMMAND_RUN_DIRECT:
+	case TM_COMMAND_STOP:
+			break;
+	default: 	return -EOPNOTSUPP;
+			break;
+	}
+
 	ev3_tm->run_command = command;
 
-	if (command == TM_COMMAND_RESET) {
-		legoev3_motor_reset(ev3_tm);
-		return 0;
-	}
+ 	/*
+	 * what's going on here - why is run always set to 1?
+	 *
+	 * the answer is that we check for run == 0 as the first condition
+	 * at the top of the timer callback. If it's set, then the next motor state
+	 * is stop_motor, but it won't be evaluated if run == 0
+	 *
+	 * so we always force the state machine to run once, and count on the
+	 * state machine to dtrt (do the right thing). This avoids setting motor
+	 * power in weird places
+	 */
 
-	if (TM_COMMAND_RUN_FOREVER == command || TM_COMMAND_RUN_DIRECT == command)
-		ev3_tm->state = STATE_RUN_FOREVER;
-	else if (IS_POS_CMD(command))
-		ev3_tm->state = STATE_SETUP_RAMP_POSITION;
+	ev3_tm->run = 1;
 
-	if (command == TM_COMMAND_RUN_TO_ABS_POS)
-		ev3_tm->position_mode = POSITION_ABSOLUTE;
-	if (command == TM_COMMAND_RUN_TO_REL_POS)
-		ev3_tm->position_mode = POSITION_RELATIVE;
+	return 0;
+}
 
-	if (TM_COMMAND_STOP == command)
-		ev3_tm->state = STATE_STOP;
+static int legoev3_motor_set_internal_command(void *context,
+			enum tacho_motor_internal_command internal_command)
+{
+	struct legoev3_motor_data *ev3_tm = context;
+
+	ev3_tm->internal_command = internal_command;
+
+	// NOTE: THIS IS REALLY tacho_motor_internal_command!!!!
```

Comment:
This note seems unnecessary because this already says "internal_command" everywhere. Or I am missing something, in which case it requires further explanation.